src/default-settings.js

```diff
@@ -1,167 +1,177 @@
-    subjectControl.value_col = preSettings.id_col;
-
-    var relatedControl = preControlInputs.filter(function(d){return d.label=="Related to Treatment"})[0];
-    relatedControl.value_col = preSettings.rel_col;
-
-    var seriousControl = preControlInputs.filter(function(d){return d.label=="Serious?"})[0];
-    seriousControl.value_col = preSettings.ser_col;
-
-    settings.filter_cols.forEach(function(d,i){
-      var thisFilter = {
-        type:"subsetter", 
-        value_col:d, 
-        label:d,
-        multiple:true
-      }
-      var filter_vars = preControlInputs.map(function(d){return d.value_col})
-      if (filter_vars.indexOf(thisFilter.value_col)== -1) {
-        preControlInputs.push(thisFilter);
-      } 
-    })
+export const controlInputs =
+    [
+        {type: 'dropdown' , option: 'y.sort', label: 'Sort IDs', values: ['earliest', 'alphabetical-descending'], require: true}
+    ];
+
+export function syncControlInputs(preControlInputs, preSettings) {
+    preSettings.filter_cols.reverse()
+        .forEach((d,i) => {
+            var thisFilter =
+                {type: 'subsetter'
+                ,value_col: d
+                ,label: d};
+            var filter_vars = preControlInputs
+                .map(d => d.value_col);
+
+          //Check whether [ filter_vars ] settings property contains default filter column.
+            if (filter_vars.indexOf(thisFilter.value_col) === -1)
+                preControlInputs.unshift(thisFilter);
+        });
   
     return preControlInputs;
 }
 
 //Setting for custom details view
-export const secondSettings = {
-    "x":{label:'', "type":"linear","column":"wc_value"},
-    "y":{label: '', "sort":"alphabetical-descending","type":"ordinal","column":"AESEQ"},
-    "marks":[
-        {"type":"line","per":["AESEQ"], attributes:{'stroke-width': 5, 'stroke-opacity': .8 }},
-        {"type":"circle","per":[ "AESEQ", "wc_value"]},
-        {"type":"line","per":[ "AESEQ"], attributes:{'stroke-width': 3, 'stroke-opacity': .8, 'stroke': 'black'},"values":{"AESER": ["Yes", 'Y']}},
-        {"type":"circle","per":[ "AESEQ", "wc_value"],"attributes":{'stroke': 'black','stroke-width': 2},"radius":5,"values":{"AESER": ["Yes", 'Y']}}
-     ],
-    "legend":{
-        "mark":"circle", 
-        "label": 'Severity'
-    },
-    "color_by": "AESEV",
-    "colors": ['#66bd63', '#fdae61', '#d73027', '#6e016b'],
-    "date_format":"%d%b%Y:%X",
-    "transitions": false,
-    "max_width":1000,
-    "gridlines":"y",
-    "no_text_size":false,
-    "range_band":28
-};
+let cloneSettings = clone(settings);
+    cloneSettings.y.sort = 'alphabetical-descending';
+    cloneSettings.transitions = false;
+    cloneSettings.range_band = settings.range_band*2;
+    cloneSettings.margin = null;
+export const secondSettings = cloneSettings;
+
+export function syncSecondSettings(preSettings) {
+    const nextSettings = Object.create(preSettings);
```

Comment:
You know, I'm still not entirely sure how to deep clone an object, like down into arrays and objects and arrays of objects and so on.  But I hate manually syncing multiple settings objects so I like this approach.  But as you can see I couldn't get the `marks` to clone :crying_cat_face: